DetectDynamicJS.py

```diff
@@ -154,15 +159,20 @@ def isScript(self, requestResponse):
             if len(contentTypeL) == 1:
                 contentType = contentTypeL[0].lower()
             statusCode = responseInfo.getStatusCode()
-            if (any(content in contentType for content in possibleContentTypes) or "script" in inferredMimeType or "script" in mimeType) and (int(statusCode) < 300 or int(statusCode) > 399):
-                return True
+            if (any(content in contentType for content in possibleContentTypes) or "script" in inferredMimeType or "script" in mimeType):
+              return True
         return False
         
 
     def compareResponses(self, newResponse, oldResponse):
         """Compare two responses in respect to their body contents"""
         nResponse = newResponse.getResponse()
         nResponseInfo = self._helpers.analyzeResponse(nResponse)
+        
+        # Check if the statusCode is still 200
```

Comment:
For reducing false positives, this is not necessary. If you get a 200 on the first request and a 302 or something afterwards you get a login oracle. So it actually creates false negatives.